lib/openssl/config.rb

```diff
@@ -52,7 +52,11 @@ def parse(string)
       # Raises a ConfigError on invalid configuration data.
       def parse_config(io)
         begin
-          parse_config_lines(io)
+          if(io.instance_of?(String))
```

Comment:
I undestand the code is in native in MRI, but would be nice to follow existing Ruby conventions
```suggestion
          if io.instance_of?(String)
```

also could you please attempt to add test for this feature, Ruby seems to have parsing tests, so they could serve as a starting point: https://github.com/ruby/openssl/blob/master/test/openssl/test_config.rb